src/python/foglamp/core/service_registry/instance.py

```diff
@@ -32,6 +32,7 @@ def __init__(self, s_id, s_name, s_type, s_protocol, s_address, s_port):
         self._protocol = s_protocol
         self._address = s_address
         self._port = int(s_port)
+        self._status = 0
```

Comment:
I think this should be a composite object, having last heartbeat time and up / down/ abort (or halted) status